src/main/java/io/anserini/search/SearchCollection.java

```diff
@@ -349,48 +345,49 @@ public void close() throws IOException {
+                RerankerCascade cascade = new RerankerCascade(tag);
                 cascade.add(new AxiomReranker(args.index, args.axiom_index, IndexArgs.CONTENTS,
                     args.axiom_deterministic, Integer.valueOf(seed), Integer.valueOf(r),
                     Integer.valueOf(n), Float.valueOf(beta), Integer.valueOf(top),
                     args.axiom_docids, args.axiom_outputQuery, args.searchtweets));
                 cascade.add(new ScoreTiesAdjusterReranker());
-                String tag = "axiom.seed:"+seed+",axiom.r:"+r+",axiom.n:"+n+",axiom.beta:"+beta+",axiom.top:"+top;
-                cascades.put(tag, cascade);
+                cascades.add(cascade);
               }
             }
           }
         }
       }
     } else if (args.bm25prf) {
-        for (String fbTerms : args.bm25prf_fbTerms) {
-            for (String fbDocs : args.bm25prf_fbDocs) {
-                for (String k1 : args.bm25prf_k1) {
-                    for (String b : args.bm25prf_b) {
-                        for (String newTermWeight : args.bm25prf_newTermWeight) {
-                            RerankerCascade cascade = new RerankerCascade();
-                            cascade.add(new BM25PrfReranker(analyzer, IndexArgs.CONTENTS, Integer.valueOf(fbTerms),
-                                    Integer.valueOf(fbDocs), Float.valueOf(k1), Float.valueOf(b), Float.valueOf(newTermWeight),
-                                    args.bm25prf_outputQuery));
-                            cascade.add(new ScoreTiesAdjusterReranker());
-                            String tag = "bm25prf.fbTerms:" + fbTerms + ",fbDocs:" + fbDocs + ",bm25prf.k1:" + k1 + ",bm25prf.b:" + b + ",bm25prf.newTermWeight:" + newTermWeight;
-                            cascades.put(tag, cascade);
-                        }
-                    }
-                }
+      for (String fbTerms : args.bm25prf_fbTerms) {
+        for (String fbDocs : args.bm25prf_fbDocs) {
+          for (String k1 : args.bm25prf_k1) {
+            for (String b : args.bm25prf_b) {
+              for (String newTermWeight : args.bm25prf_newTermWeight) {
+                String tag = String.format("bm25prf(fbTerms=%s,fbDocs=%s,k1=%s,b=%s,newTermWeight=%s)",
+                    fbTerms, fbDocs, k1, b, newTermWeight);
+                RerankerCascade cascade = new RerankerCascade(tag);
+                cascade.add(new BM25PrfReranker(analyzer, IndexArgs.CONTENTS, Integer.valueOf(fbTerms),
+                    Integer.valueOf(fbDocs), Float.valueOf(k1), Float.valueOf(b), Float.valueOf(newTermWeight),
+                    args.bm25prf_outputQuery));
+                cascade.add(new ScoreTiesAdjusterReranker());
+                cascades.add(cascade);
+              }
             }
+          }
         }
+      }
     } else {
-      RerankerCascade cascade = new RerankerCascade();
+      RerankerCascade cascade = new RerankerCascade("default");
```

Comment:
Since we use the "default" tag a lot, maybe add a constructor with no arguments that automatically sets the tag as "default"? Then we can just use `new RerankerCascade()` here and elsewhere

Reply:
👍